backend-implicaction/src/main/java/com/dynonuggets/refonteimplicaction/service/JobPostingService.java

```diff
@@ -53,4 +55,19 @@ public void deleteJobPosting(Long jobPostingId) {
                 .orElseThrow(() -> new NotFoundException("Impossible de supprimer l'offre, " + jobPostingId + " n'existe pas."));
         jobPostingRepository.delete(jobPosting);
     }
+
+    @Transactional
+    public JobPostingDto archiveOrUnarchiveJobPosting(Long jobPostingId) {
+        JobPosting jobPosting = jobPostingRepository.findById(jobPostingId)
+                .orElseThrow(() -> new NotFoundException("Impossible de supprimer l'offre, " + jobPostingId + " n'existe pas."));
```

Comment:
La classe message a été créée pour stocker ce genre de message. Tu peux créer une constante dans cette classe stp ? De plus attention à ton copier coller, tu cherches pas à supprimer une offre ;)

```suggestion
                .orElseThrow(() -> new NotFoundException(String.format(Message.JOB_NOT_FOUND, jobId)));
```